handler/src/main/java/io/netty/handler/ssl/OpenSsl.java

```diff
@@ -72,45 +72,54 @@
     static {
         Throwable cause = null;
 
-        // Test if netty-tcnative is in the classpath first.
-        try {
-            Class.forName("io.netty.internal.tcnative.SSL", false, OpenSsl.class.getClassLoader());
-        } catch (ClassNotFoundException t) {
-            cause = t;
-            logger.debug(
-                    "netty-tcnative not in the classpath; " +
-                    OpenSslEngine.class.getSimpleName() + " will be unavailable.");
-        }
+        if (SystemPropertyUtil.getBoolean("io.netty.handler.ssl.openssl.disabled", false)) {
+            cause = new UnsupportedOperationException(
+                    "OpenSSL was explicit disabled with -Dio.netty.handler.ssl.openssl.disabled=true");
 
-        // If in the classpath, try to load the native library and initialize netty-tcnative.
-        if (cause == null) {
+            logger.debug(
+                    "netty-tcnative explicit disabled; " +
+                            OpenSslEngine.class.getSimpleName() + " will be unavailable.", cause);
+        } else {
+            // Test if netty-tcnative is in the classpath first.
             try {
-                // The JNI library was not already loaded. Load it now.
-                loadTcNative();
-            } catch (Throwable t) {
+                Class.forName("io.netty.internal.tcnative.SSL", false, OpenSsl.class.getClassLoader());
+            } catch (ClassNotFoundException t) {
                 cause = t;
                 logger.debug(
-                    "Failed to load netty-tcnative; " +
-                        OpenSslEngine.class.getSimpleName() + " will be unavailable, unless the " +
-                        "application has already loaded the symbols by some other means. " +
-                        "See http://netty.io/wiki/forked-tomcat-native.html for more information.", t);
+                        "netty-tcnative not in the classpath; " +
+                                OpenSslEngine.class.getSimpleName() + " will be unavailable.");
```

Comment:
nit: include `t` in the debug?

Reply:
I can do this as a followup if you like as this code was just moved.